CvGameCoreDLL_Expansion2/CvUnit.cpp

```diff
@@ -31260,73 +31259,160 @@ int CvUnit::GetWithdrawChance(const CvUnit& attacker, const bool bCheckChances)
 
 		if (pDestPlot && !canMoveInto(*pDestPlot, MOVEFLAG_DESTINATION | MOVEFLAG_NO_EMBARK))
 		{
-			iBlockedHexes++;
+			iFreeHexes--;
 		}
 	}
 
-	// If all three hexes away from attacker blocked, we can't withdraw
-	if (iBlockedHexes >= 3)
+	return iFreeHexes;
+}
+
+int CvUnit::GetWithdrawChance(const CvUnit& attacker) const
+{
+	VALIDATE_OBJECT
+	int iWithdrawChance = getExtraWithdrawal();
+	if (iWithdrawChance == 0)
 		return 0;
-	
-	if (bCheckChances)
+
+	if (GetNumFallBackPlotsAvailable(attacker) == 0)
```

Comment:
Given the work performed in `GetNumFallbackPlotsAvailable()`, this should be set to a variable, so it can be used without rerunning it again on line 31291.

Reply:
Done.